docs/docsite/rst/network_declarative_intent.rst

```diff
@@ -0,0 +1,349 @@
+*********************************
+Declarative intent in networking
+*********************************
+
+.. contents:: Topics
+
+This section explores what `declarative intent modules` are within the context of Ansible networking.
+
+This document is part of a collection on networking. The complete list of guides can be found at :ref:`network guides <guide_networking>`.
+
+Overview
+========
+
+INTERNAL NOTE: TBD - Short description of the problem. Needs to be very simple
+
+State what you want the end result to be, not how go get there
+This results in easier to understand playbooks, which internally deal with the differences between network OS versions.
+
+
+Reminder: Ansible isn't a monitoring tool.
+
+
+
+Without declarative intent
+==========================
+
+INTERNAL NOTE: TBD - Simplist case that shows the issue. We can build on this later
+
+Enable a port and set vlan
+
+.. code-block:: yaml
+
+   - name: Enable port
+
+Ansible reports everything is OK.
+
+The issue here is that we are only asserting *configuration* changes, there is no checks of the *state* of the network infrastructure, for example has the cable been plugged in?
+
+
+With declaritive intent
+=======================
+
+INTERNAL NOTE: TBD - Simplist case that shows the solution. We can build out the example in the later use-cases.
+
+**Simple example**
+
+.. code-block:: yaml
+
+   - name: configure interface
+     net_interface:
+       # Declared configuration
+       name: GigabitEthernet0/2
+       description: public interface configuration
+       enabled: yes
+
+       # Intended state
+       state: connected
```

Comment:
The intended state value is up/down [reference](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/interface/net_interface.py#L70)